src/components/app/App.js

```diff
@@ -1,11 +1,33 @@
-import React from 'react';
+import React, {createContext, useReducer} from 'react';
 import '../../styles/App.scss';
+import  Home from '../home/Home';
+import CenterContainer from '../votingcenter/CenterContainer'
 
+const initialState = {
+  address: null
+}
+
+const reducer = (state, action) => {
+  switch(action.type){
+    case "submitAddress":
+      return {...state,  address: action.payload}
+    default:
+      return state;
+  }
+}
+        
 function App() {
+  const [state, dispatch] = useReducer(reducer, initialState)
+
   return (
     <div className="App">
+      <AppContext.Provider value={{state, dispatch}}>
+        <Home/>
+        <CenterContainer/>
+      </AppContext.Provider>
```

Comment:
We wrap the AppContext around the children components, similar to how we wrap the react-redux Provider around the App when using redux

Reply:
also the value that we pass down is an object with the state obj and dispatch function inside it